orderbook/src/main.rs

```diff
@@ -434,3 +436,84 @@ async fn check_database_connection(orderbook: &Orderbook) {
         .await
         .expect("failed to connect to database");
 }
+
+/// Parses a comma separated list of colon separated values representing fee factors for AppIds.
+fn parse_partner_fee_factor(s: &str) -> Result<HashMap<AppId, f64>> {
+    let mut res = HashMap::default();
+    for pair_str in s.split(',').into_iter().collect::<Vec<&str>>() {
+        let split_pair = pair_str
+            .trim()
+            .split(':')
+            .into_iter()
+            .map(|x| x.trim())
+            .collect::<Vec<&str>>();
```

Comment:
Not very important but we could also avoid allocating a vector here by keeping this as an iterator and doing `.next()` twice manually and failing if next returns none. Then call next a third time and fail if it isn't None.